extensions/sql/contract-definition/store/src/test/java/org/eclipse/dataspaceconnector/sql/contractdefinition/store/SqlContractDefinitionStoreTest.java

```diff
@@ -0,0 +1,214 @@
+        contractDefinitionStore.save(definition);
+
+        var definitions = contractDefinitionStore.findAll();
+
+        assertThat(definitions).isNotNull();
+        assertThat(definitions.size()).isEqualTo(1);
+    }
+
+    @Test
+    @DisplayName("Save multiple Contract Definitions")
+    void saveMany() {
+        var definitionsCreated = getContractDefinitions(10);
+        contractDefinitionStore.save(definitionsCreated);
+
+        var definitionsRetrieved = contractDefinitionStore.findAll();
+
+        assertThat(definitionsRetrieved).isNotNull();
+        assertThat(definitionsRetrieved.size()).isEqualTo(definitionsCreated.size());
+    }
+
+    @Test
+    @DisplayName("Update a Contract Definition")
+    void updateOne() throws SQLException {
+        var definition1 = getContractDefinition("id", "contract1", "policy1");
+        var definition2 = getContractDefinition("id", "contract2", "policy2");
+
+        contractDefinitionStore.save(definition1);
+        contractDefinitionStore.update(definition2);
+
+        var query = String.format("SELECT * FROM %s WHERE %s=?",
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_ID);
+
+        var definitions = executeQuery(dataSourceRegistry.resolve(ConfigurationKeys.DATASOURCE_NAME).getConnection(),
+                ((SqlContractDefinitionStore) contractDefinitionStore)::mapResultSet,
+                query,
+                definition1.getId());
+
+        assertThat(definitions).isNotNull();
+        assertThat(definitions.size()).isEqualTo(1);
+        assertThat(definitions.get(0).getContractPolicy().getUid()).isEqualTo(definition2.getContractPolicy().getUid());
+        assertThat(definitions.get(0).getAccessPolicy().getUid()).isEqualTo(definition2.getAccessPolicy().getUid());
+    }
+
+    @Test
+    @DisplayName("Find all contract definitions")
+    void findAll() {
+        var definitionsExpected = getContractDefinitions(10);
+        contractDefinitionStore.save(definitionsExpected);
+
+        var definitionsRetrieved = contractDefinitionStore.findAll();
+
+        assertThat(definitionsRetrieved).isNotNull();
+        assertThat(definitionsRetrieved.size()).isEqualTo(definitionsExpected.size());
+    }
+
+    @Test
+    @DisplayName("Find all contract definitions with limit and offset")
+    void findAllWithSpec() {
+        var limit = 20;
+
+        var definitionsExpected = getContractDefinitions(50);
+        contractDefinitionStore.save(definitionsExpected);
+
+        var spec = QuerySpec.Builder.newInstance()
+                .limit(limit)
+                .offset(20)
+                .build();
+
+        var definitionsRetrieved = contractDefinitionStore.findAll(spec).collect(Collectors.toList());
+
+        assertThat(definitionsRetrieved).isNotNull();
+        assertThat(definitionsRetrieved.size()).isEqualTo(limit);
+    }
```

Comment:
again, this test misses quite a few test cases. `update_whenNotExists`, `saveAll_whenSomeExist`, `saveAll_whenAllExist`, etc.